LoopKitUI/Views/SegmentedGaugeBarView.swift

```diff
@@ -96,26 +112,109 @@ public class SegmentedGaugeBarView: UIView {
         setupPanGestureRecognizer()
     }
 
+    private lazy var thumb = ThumbView()
+
+    public var displaysThumb = false {
+        didSet {
+            if displaysThumb {
+                assert(numberOfSegments == 1, "Thumb only supported for single-segment gauges")
+                if thumb.superview == nil {
+                    addSubview(thumb)
+                }
+            } else {
+                thumb.removeFromSuperview()
+            }
+        }
+    }
+
+    private var panGestureRecognizer: UIPanGestureRecognizer?
+
     private func setupPanGestureRecognizer() {
         let pan = UIPanGestureRecognizer(target: self, action: #selector(handlePan(_:)))
+        panGestureRecognizer = pan
         addGestureRecognizer(pan)
     }
 
     @objc private func handlePan(_ recognizer: UIPanGestureRecognizer) {
         switch recognizer.state {
         case .began, .changed:
+            guard recognizer.numberOfTouches == 1 else {
+                break
+            }
+
             let location = recognizer.location(ofTouch: 0, in: self)
-            let fractionThrough = Double(location.x / frame.width)
+            let gaugeFillFraction = fractionThrough(location.x, in: gaugeXRange)
             let oldValue = progress
-            progress = (fractionThrough * Double(numberOfSegments)).clamped(to: 0...Double(numberOfSegments))
-            delegate?.segmentedGaugeBarView(self, didUpdateProgressFrom: oldValue, to: progress)
+            let newValue = (Double(gaugeFillFraction) * Double(numberOfSegments)).clamped(to: 0...Double(numberOfSegments))
+            CATransaction.withoutActions {
+                progress = newValue
+            }
+
+            delegate?.segmentedGaugeBarView(self, didUpdateProgressFrom: oldValue, to: newValue)
+        case .ended:
+            uglyWorkaroundToForceRedraw()
         default:
             break
         }
     }
 
+    private func uglyWorkaroundToForceRedraw() {
```

Comment:
I'm far from thrilled with this hack, but it was the best I could come up with to force CoreAnimation's presentation layer to resync with the model layer. Probably should reproduce this in a sample project to file a radar, but I spent a long time banging my head on this one already..